src/protocols/ethereum/libs/EthereumAdapter.ts

```diff
@@ -217,6 +220,94 @@ export class EthereumAdapter extends BaseProtocolAdapter {
     }
   }
 
+  override async fetchTokenInfo(contractId: string): Promise<IToken | undefined> {
+    const { ethActiveNetworkPredefinedSettings } = useEthNetworkSettings();
+    const apiUrl = ethActiveNetworkPredefinedSettings.value.tokenMiddlewareUrl;
+    try {
+      // Temporary solution for fetching the ERC-20 token info.
+      // TODO Replace with our own node API
+      return new EthplorerService(apiUrl).fetchTokenInfo(contractId);
+    } catch (error: any) {
+      Logger.write(error);
+      return undefined;
+    }
+  }
+
+  override async transferToken(
+    amount: number,
+    recipient: string,
+    contractId: string,
+    options: {
+      fromAccount: string;
+      maxPriorityFeePerGas: string;
+      maxFeePerGas: string;
+    },
+  ): Promise<ITransferResponse> {
+    const {
+      ethActiveNetworkSettings,
+      ethActiveNetworkPredefinedSettings,
+    } = useEthNetworkSettings();
+    const { getAccountByAddress } = useAccounts();
+    const apiUrl = ethActiveNetworkPredefinedSettings.value.middlewareUrl;
+
+    const contractAbi = (await new EtherscanService(apiUrl)
+      .fetchFromApi({
+        module: 'contract',
+        action: 'getabi',
+        address: contractId,
+      }));
+
+    const contract = new Contract(
+      contractAbi && Array.isArray(contractAbi?.result) ? contractAbi.result : ERC20_ABI,
+      contractId,
+      { from: options.fromAccount },
+    );
+
+    const { chainId, nodeUrl } = ethActiveNetworkSettings.value;
+    contract.setProvider(nodeUrl);
+
+    const hexAmount = bigIntToHex(BigInt(toWei(amount.toFixed(
+      Number(await contract.methods.decimals().call()),
+    ), 'ether')));
+    const maxPriorityFeePerGas = bigIntToHex(BigInt(toWei(options.maxPriorityFeePerGas, 'ether')));
+    const maxFeePerGas = bigIntToHex(BigInt(toWei(options.maxFeePerGas, 'ether')));
+
+    const [nonce, gasLimit] = await Promise.all([
+      this.getTransactionCount(options.fromAccount),
+      // @ts-expect-error
+      contract.methods.transfer(recipient, hexAmount).estimateGas(),
+    ]);
+
+    // All values are in wei
+    const txData: FeeMarketEIP1559TxData = {
+      chainId,
+      nonce,
+      to: contractId,
+      // @ts-expect-error
+      data: contract.methods.transfer(recipient, hexAmount).encodeABI(),
+      value: 0x0,
+      maxPriorityFeePerGas,
+      maxFeePerGas,
+      gasLimit: `0x${gasLimit.toString(16)}`,
+      type: '0x02',
+    };
+
+    const tx = FeeMarketEIP1559Transaction.fromTxData(txData);
+
+    const account = getAccountByAddress(options.fromAccount);
+    if (!account || account.protocol !== PROTOCOLS.ethereum) {
+      throw new Error('Token transfer were initiated from not existing or not ethereum account.');
+    }
+
+    const signedTx = tx.sign(account.secretKey!);
+
+    const serializedTx = signedTx.serialize();
+    const web3Eth = this.getWeb3EthInstance();
+    const res = await sendSignedTransaction(web3Eth, serializedTx, DEFAULT_RETURN_FORMAT);
```

Comment:
I suggest to take a look at [this PR](https://github.com/superhero-com/superhero-wallet/pull/2525/files)
where I'm calculating the `hash` from the `serializedTx` instead of waiting for the transaction and then getting the `hash` from the `res`.
Judging by the time it takes `sendSignedTransaction` to resolve, it is probably waiting for some confirmation which takes a lot of time.